pennylane/interfaces/autograd.py

```diff
@@ -68,32 +82,15 @@ def gradient_product(g):
                     nested Sequence[float]: vector-Jacobian product, arranged
                     into the nested structure of the input arguments in ``args``
                 """
-                diff_indices = None
-                non_diff_indices = set()
-
-                for arg, arg_variable in zip(args, self.arg_vars):
-                    if not getattr(arg, "requires_grad", True):
-                        indices = [i.idx for i in _flatten(arg_variable)]
-                        non_diff_indices.update(indices)
-
-                if non_diff_indices:
-                    diff_indices = set(range(self.num_variables)) - non_diff_indices
-
                 # Jacobian matrix of the circuit
-                jac = self.jacobian(args, kwargs, wrt=diff_indices)
+                self.set_trainable(args)
+                jac = self.jacobian(args, kwargs)
 
                 if not g.shape:
                     vjp = g * jac  # numpy treats 0d arrays as scalars, hence @ cannot be used
                 else:
                     vjp = g @ jac
 
-                if non_diff_indices:
-                    # Autograd requires we return a gradient of size (num_variables,)
-                    res = zeros([self.num_variables])
-                    indices = fromiter(diff_indices, dtype=int64)
-                    res[indices] = vjp
-                    vjp = res
-
```

Comment:
How come this conversion is no longer needed?

Reply:
See https://github.com/XanaduAI/pennylane/pull/660/files#r433597583